### DISCUSSION TITLE
The database.rdb file is been modified frequently.

### DISCUSSION THREAD

**Discussion by gth7754125** [Difegue/LANraragi#462] [Q&A] [answered]
**LRR Version and OS**
Docker version 0.78 Newest

**Bug Details**
Although is not a bug, but i am curious what job or task Lanraragi execute to make the redis's rdb files update frequently.
I put the rdb files to Online cloud drive as a backup. When i run Lanraragi on my PC 1 (only run as usual), my PC 2, which  can find the rdb files keep upload to cloud drive because it is modified so that it trigger the cloud drive storage mechanism.
Around few seconds, the rdb file will be updated.
Maybe put the rdb file to cloud drive is not a good idea to backup the metadata ? ..QAQ

**Matching Logs**
None.

**Screenshots**
None.

**Comment by Difegue** [comment, marked as answer]
The .rdb is, as its name implies, the database, so anytime a change occurs to it the .rdb file gets modified/saved.  

Technically speaking, Redis has options to limit the rate of writes to disk, but I've chosen to maximize them for the Docker containers to avoid potential data loss issues.
(See https://github.com/Difegue/LANraragi/blob/5fb4e11ec360c44b0edb2c7b7d443185285c047d/tools/build/docker/redis.conf#L218 )

I suppose you could overwrite the container's config with a custom one through a bind mount to change this parameter to something that saves less often, if it really is an issue.

**Reply by gth7754125** [reply]
Eh...i saw #359 . Actually, if both cache and persistant data use redis, we can divide them into two db in redis, like db0 and db1. And only db1 need to be persistance.   
As for the **save 1 1**, i think 1 1 is extreme. I know it's not convenient to maintain  two database like sqllite and redis, so why don't we just persistent important data to a binary or json file ?  If the data is sensitive, why not use aof. Normally we just use it as distributed cache. And apparently there is no need to use aof, the same as rdb mode which set **save 1 1**. Also I don't know whether too frequently to do persistent job will affect performance on Lanraragi.

**Reply by koma-cute** [reply]
Keep in mind `save 1 1` still only writes to disk if any changes are actually made, so if your instance is sitting dormant, it won't write to disk. If you want a json backup, you can use the webui or API endpoint `/api/backup` at your leisure, while disabling automatic syncing of the Redis DB to your cloud storage. AOF could be an advantage (as well as using a real DB), but you'd still be hitting your disk just as often.
